docs/API/v3/registration-base-url-resource.md

```diff
@@ -0,0 +1,301 @@
+any version of the dependency is allowed.
+
+#### Example
+
+The registration index for NuGet.Server.Core on nuget.org looks something like this:
+
+[!code-JSON [package-registration-index.json](./_data/package-registration-index.json)]
+
+In this particular case, the registration index has the registration page inlined so no extra requests are needed to
+fetch metadata about individual package versions.
+
+## Registration page
+
+The registration page contains registration leaves. The URL to fetch a registration page is determined by the `@id`
+property in the [registration page object](#registration-page-object) mentioned above.
+
+When the `items` array is not provided in the registration index, an HTTP GET request of the `@id` value will return a
+JSON document which has an object as its root. The object has the following properties:
+
+Name   | Type             | Required | Notes
+------ | ---------------- | -------- | -----
+@id    | string           | yes      | The URL to the registration page
+count  | integer          | yes      | The number of registration leaves in the page
+items  | array of objects | yes      | The array of registration leaves and their associate metadata
+lower  | string           | yes      | The lowest SemVer 2.0.0 version in the page (inclusive)
+parent | string           | yes      | The URL to the registration index
+upper  | string           | yes      | The highest SemVer 2.0.0 version in the page (inclusive)
+
+The shape of the registration leaf objects is the same as in the registration index
+[above](#registration-leaf-object-in-a-page).
+
+## Registration leaf
+
+The registration leaf contains information about a specific package ID and version. The metadata about the specific
+version may not be available in this document. Package metadata should be fetched from the
+[registration index](#registration-index) or the [registration page](#registration-page) (which is discovered using
+the registration index).
+
+The URL to fetch a registration leaf is obtain from the `@id` property of a registration leaf object in either a
+registration index or registration page.
+
+The registration leaf is a JSON document with a root object with the following properties:
+
+Name           | Type    | Required | Notes
+-------------- | ------- | -------- | -----
+@id            | string  | yes      | The URL to the registration leaf
+catalogEntry   | string  | no       | The URL to the catalog entry that produced these leaf
+listed         | boolean | no       | Should be considered as listed if absent
+packageContent | string  | no       | The URL to the package content (.nupkg)
+published      | string  | no       | A string containing a ISO 8601 timestamp of when the package was published
+registration   | string  | no       | The URL to the registration index
+
+> [!Note]
+> On nuget.org, the `published` value is set to year 1900 when the package is unlisted.
```

Comment:
To be precise, isn't it `1900-01-01`?

Reply:
There is some inconsistency in our implementation:
https://github.com/NuGet/NuGet.Services.Metadata/blob/4027fa0a465d1927f42a3cab56bb947872af92dd/src/Catalog/Registration/RegistrationMakerCatalogItem.cs#L125

For this reason, I'd like to leave the documents more generic.